emily/lambda/src/main.rs

```diff
@@ -0,0 +1,175 @@
+use aws_lambda_events::apigw::{ApiGatewayProxyRequest, ApiGatewayProxyResponse};
+use aws_lambda_events::encodings::Body;
+use emily::models::{CreateDepositRequestContent, DepositParameters};
+use http::{header, HeaderValue};
+use lambda_runtime::{service_fn, Error, LambdaEvent};
+use http::{HeaderMap, Method};
+
+/// Asynchronously handles incoming API Gateway events and dispatches them
+/// to the correct method based on the route and method.
+///
+/// # Arguments
+/// * `event` - A LambdaEvent containing the ApiGatewayProxyRequest.
+///
+/// # Returns
+/// A result containing the API Gateway Proxy Response or an error.
+async fn handle_event(
+    event: LambdaEvent<ApiGatewayProxyRequest>
+) -> Result<ApiGatewayProxyResponse, Error> {
+
+    // Extract base data.
+    let resource = event.payload.resource.unwrap_or_default();
+    let http_method = event.payload.http_method;
+    let body = event.payload.body.clone();
+
+    // Dispatch based on API call.
+    match (resource.as_str(), http_method) {
+        ("/deposits", Method::POST) => {
+
+            // Deserialize the create deposit request.
+            let request: CreateDepositRequestContent = body
+                .map(|serailzed_request| {
+                    serde_json::from_str::<CreateDepositRequestContent>(serailzed_request.as_str())
+                    .unwrap() // TODO: Handle errors.
```

Comment:
Is this to be done in the current PR? If you want to do it as a follow-up I'd like to have the todo comment reference an issue as `// TODO(<issue_number>): Handle errors`. In general, I don't want any unlabeled todos to make it to the main branch.

Reply:
Yes, that's generally fair. I'd like to leave it for now because I'm about to wipe the entire file, this is just to get something working and to remind myself that that part was dubious so when I rewrite the file I can handle those errors.

I'm 100% in on linking all TODOs to tickets, but in this case this code is going to have a shelf life of around 2 or so days and the GitHub is going to be come out of date nearly immediately.

Reply:
Actually, I just created one of the implementation tickets #111 and linked the TODOS to it. It's not directly related but it's the ticket I'll address it in.

Reply:
Done.

Reply:
Nice, thanks :pray: